### DISCUSSION TITLE
RedirectToCanonicalUrlAttribute converts query strings to lower case

### DISCUSSION THREAD

**Discussion by claq2** [ASP-NET-MVC-Boilerplate/Framework#15] [Q&A]
Query string parameters shouldn't be altered. Some systems, like OAuth2, expect values to be reflected as-is. (.NET code doesn't change them when you set RouteCollection.LowercaseUrls to true, only the path.)

Please make RedirectToCanonicalUrlAttribute not change query string parameters.

**Comment by RehanSaeed** [comment]
The query string is [case sensitive](https://webmasters.stackexchange.com/questions/88586/do-query-string-case-differences-affect-whether-a-url-is-canonical) for SEO. In my experience it's fairly rare to want to use a case sensitive query string but when you do, use `[NoLowercaseQueryString]`.
